### PR TITLE
feat(encrypt_dir): Add and change current

### PR DESCRIPTION
**Pull request by henrikstranneheim**
## Description

Separate encryption dir for backup retrieval and, encryption and backup archiving. Since retrieval  has one more path level than encrypting and archiving.

### Added

- Encryption dir

### Changed

- Separate encryption dir for backup retrieval and, encryption and backup archiving. Since retrieval  has one more path level than encrypting and archiving.

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b fix-current-encryption-dir -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [x] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

**Comment by henrikstranneheim** [comment]
Linked to: https://github.com/Clinical-Genomics/servers/pull/1195

File: cg/models/cg_config.py [hunk 1]
```diff
@@ -42,6 +42,11 @@ class SlurmConfig(BaseModel):
...
 
 
+class Encryption(BaseModel):
+    encryption_dir: str
+    binary_path: str
+
+
 class EncryptionDirectories(BaseModel):
     current: str
     nas: str
```
Comment (ChrOertlin) [inline cg/models/cg_config.py:52]: Do they relate to each other in some way? Are both needed?

File: tests/cli/backup/test_backup_command.py [hunk 2]
```diff
@@ -30,7 +30,7 @@ def test_backup_flow_cells(
     )
 
     # GIVEN an encrypted flow cell
-    flow_cells_dir = Path(cg_context.backup.encryption_directories.current, flow_cell_full_name)
+    flow_cells_dir = Path(cg_context.encryption.encryption_dir, flow_cell_full_name)
```
Comment (ChrOertlin) [inline tests/cli/backup/test_backup_command.py:33]:
```suggestion
    flow_cells_encrypt_dir = Path(cg_context.encryption.encryption_dir, flow_cell_full_name)
```
flow_cells_dir == /home/proj/production/flow_cells   - might be confusing at some point, but can also be argued that it is isolated within the function

File: tests/conftest.py [hunk 3]
```diff
@@ -3284,7 +3287,7 @@ def store_with_case_and_sample_with_reads(
     downsample_sample_internal_id_2: str,
 ) -> Store:
     """Return a store with a case and a sample with reads."""
-    case: Family = helpers.add_case(
+    case: Case = helpers.add_case(
```
Comment (ChrOertlin) [inline tests/conftest.py:3290]: nice spot!

**Review by ChrOertlin — APPROVED** [review]
💯 looks good!

File: cg/models/cg_config.py (hunk 1 above)
Comment (henrikstranneheim) [reply cg/models/cg_config.py]: Done

File: tests/cli/backup/test_backup_command.py (hunk 2 above)
Comment (henrikstranneheim) [reply tests/cli/backup/test_backup_command.py]: Fixed

File: tests/conftest.py (hunk 3 above)
Comment (henrikstranneheim) [reply tests/conftest.py:3293]: Ruff did it for me 😄

**Comment by henrikstranneheim** [comment]
@ChrOertlin Ready for another review!

**Comment by henrikstranneheim** [comment]
### cg backup encrypt-flow-cell ✅ 

```
θ65° [henrik.stranneheim@hasta:~] [S_base] 11s 130 $ cg --log-level DEBUG backup encrypt-flow-cells -d
Instantiating status db
Search for flow cells ready to encrypt in /home/proj/stage/flow_cells
Found directory: /home/proj/stage/flow_cells/230912_A00621_0908_AHJWJ5DSX7
Instantiating FlowCellDirectoryData with path /home/proj/stage/flow_cells/230912_A00621_0908_AHJWJ5DSX7
Set flow cell id to AHJWJ5DSX7
Using BCL converter: dragen
Found directory: /home/proj/stage/flow_cells/230908_A01901_0231_BHJWJFDSX7
Instantiating FlowCellDirectoryData with path /home/proj/stage/flow_cells/230908_A01901_0231_BHJWJFDSX7
Set flow cell id to BHJWJFDSX7
Using BCL converter: dragen
Found directory: /home/proj/stage/flow_cells/230907_A00621_0903_AHJT27DRX3
Instantiating FlowCellDirectoryData with path /home/proj/stage/flow_cells/230907_A00621_0903_AHJT27DRX3
Set flow cell id to AHJT27DRX3
Using BCL converter: dragen
Found directory: /home/proj/stage/flow_cells/170531_D00410_0425_AHKL7LBCXY
Instantiating FlowCellDirectoryData with path /home/proj/stage/flow_cells/170531_D00410_0425_AHKL7LBCXY
Set flow cell id to AHKL7LBCXY
Using BCL converter: bcl2fastq
Found directory: /home/proj/stage/flow_cells/190927_A00689_0069_BHLYWYDSXX
Instantiating FlowCellDirectoryData with path /home/proj/stage/flow_cells/190927_A00689_0069_BHLYWYDSXX
Set flow cell id to BHLYWYDSXX
Using BCL converter: dragen
Found directory: /home/proj/stage/flow_cells/20230824_LH00217_0004_A225CW7LT3
Instantiating FlowCellDirectoryData with path /home/proj/stage/flow_cells/20230824_LH00217_0004_A225CW7LT3
Set flow cell id to A225CW7LT3
Using BCL converter: dragen
Found directory: /home/proj/stage/flow_cells/230904_A00621_0900_AHJT7VDRX3
Instantiating FlowCellDirectoryData with path /home/proj/stage/flow_cells/230904_A00621_0900_AHJT7VDRX3
Set flow cell id to AHJT7VDRX3
Using BCL converter: dragen
Found directory: /home/proj/stage/flow_cells/230824_A00187_0994_BHGW2VDRX3
Instantiating FlowCellDirectoryData with path /home/proj/stage/flow_cells/230824_A00187_0994_BHGW2VDRX3
Set flow cell id to BHGW2VDRX3
Using BCL converter: dragen
Found directory: /home/proj/stage/flow_cells/20230427_LH00188_0001_B223YYCLT3
Instantiating FlowCellDirectoryData with path /home/proj/stage/flow_cells/20230427_LH00188_0001_B223YYCLT3
Set flow cell id to B223YYCLT3
Using BCL converter: dragen
Found directory: /home/proj/stage/flow_cells/20230508_LH00188_0003_A22522YLT3
Instantiating FlowCellDirectoryData with path /home/proj/stage/flow_cells/20230508_LH00188_0003_A22522YLT3
Set flow cell id to A22522YLT3
Using BCL converter: dragen
Found directory: /home/proj/stage/flow_cells/230622_A00621_0864_BH5F5YDSX7
Instantiating FlowCellDirectoryData with path /home/proj/stage/flow_cells/230622_A00621_0864_BH5F5YDSX7
Set flow cell id to BH5F5YDSX7
Using BCL converter: dragen
Found directory: /home/proj/stage/flow_cells/230810_A00689_0860_BHGNVGDRX3
Instantiating FlowCellDirectoryData with path /home/proj/stage/flow_cells/230810_A00689_0860_BHGNVGDRX3
Set flow cell id to BHGNVGDRX3
Using BCL converter: dragen
Found directory: /home/proj/stage/flow_cells/230912_A00187_1009_AHK33MDRX3
Instantiating FlowCellDirectoryData with path /home/proj/stage/flow_cells/230912_A00187_1009_AHK33MDRX3
Set flow cell id to AHK33MDRX3
Using BCL converter: dragen
Found directory: /home/proj/stage/flow_cells/221019_A00689_0663_BH7FYHDRX2
Instantiating FlowCellDirectoryData with path /home/proj/stage/flow_cells/221019_A00689_0663_BH7FYHDRX2
Set flow cell id to BH7FYHDRX2
Using BCL converter: dragen
Found directory: /home/proj/stage/flow_cells/200207_A00689_0103_BHYG7YDSXX
Instantiating FlowCellDirectoryData with path /home/proj/stage/flow_cells/200207_A00689_0103_BHYG7YDSXX
Set flow cell id to BHYG7YDSXX
Using BCL converter: dragen
Found directory: /home/proj/stage/flow_cells/170530_ST-E00214_0154_AHJCL5ALXX
Instantiating FlowCellDirectoryData with path /home/proj/stage/flow_cells/170530_ST-E00214_0154_AHJCL5ALXX
Set flow cell id to AHJCL5ALXX
Using BCL converter: bcl2fastq
Flow cell: HJWJ5DSX7 is already backed-up
Flow cell: HJWJFDSX7 is already backed-up
Flow cell: HJT27DRX3 is already backed-up
Flow cell: HKL7LBCXY is already backed-up
Flow cell: HLYWYDSXX is already backed-up
Flow cell: 225CW7LT3 is already backed-up
Flow cell: HJT7VDRX3 is already backed-up
Flow cell: HGW2VDRX3 is already backed-up
Initialising Process with binary: sbatch
Use base call ['sbatch']
Initialising Process with binary: /bin/tar
Use base call ['/bin/tar']
Initialising Process with binary: /bin/gpg
Use base call ['/bin/gpg']
SlurmAPI: Set dry run to True
Check if flow cell is ready for downstream processing
Check if sequencing is done
Sequence is done for flow cell 223YYCLT3
Check if copy of data from sequence instrument is ready
All data has been transferred for flow cell 223YYCLT3
Flow cell 223YYCLT3 is ready for downstream processing
Encryption already completed for flow cell: 223YYCLT3
Initialising Process with binary: sbatch
Use base call ['sbatch']
Initialising Process with binary: /bin/tar
Use base call ['/bin/tar']
Initialising Process with binary: /bin/gpg
Use base call ['/bin/gpg']
SlurmAPI: Set dry run to True
Check if flow cell is ready for downstream processing
Check if sequencing is done
Sequence is done for flow cell 22522YLT3
Check if copy of data from sequence instrument is ready
All data has been transferred for flow cell 22522YLT3
Flow cell 22522YLT3 is ready for downstream processing
Encryption already completed for flow cell: 22522YLT3
Flow cell: H5F5YDSX7 is already backed-up
Flow cell: HGNVGDRX3 is already backed-up
Flow cell: HK33MDRX3 is already backed-up
Flow cell: H7FYHDRX2 is already backed-up
Initialising Process with binary: sbatch
Use base call ['sbatch']
Initialising Process with binary: /bin/tar
Use base call ['/bin/tar']
Initialising Process with binary: /bin/gpg
Use base call ['/bin/gpg']
SlurmAPI: Set dry run to True
Check if flow cell is ready for downstream processing
Check if sequencing is done
Sequencing is not completed for flow cell HYG7YDSXX
Flow cell: HYG7YDSXX is not ready
Flow cell: HJCL5ALXX is already backed-up
θ63° [henrik.stranneheim@hasta:~] [S_base] 20s $ ls /home/proj/stage/flow_cells/
170530_ST-E00214_0154_AHJCL5ALXX  20230427_LH00188_0001_B223YYCLT3  230622_A00621_0864_BH5F5YDSX7  230907_A00621_0903_AHJT27DRX3
170531_D00410_0425_AHKL7LBCXY     20230508_LH00188_0003_A22522YLT3  230810_A00689_0860_BHGNVGDRX3  230908_A01901_0231_BHJWJFDSX7
190927_A00689_0069_BHLYWYDSXX     20230824_LH00217_0004_A225CW7LT3  230824_A00187_0994_BHGW2VDRX3  230912_A00187_1009_AHK33MDRX3
200207_A00689_0103_BHYG7YDSXX     221019_A00689_0663_BH7FYHDRX2     230904_A00621_0900_AHJT7VDRX3  230912_A00621_0908_AHJWJ5DSX7
θ65° [henrik.stranneheim@hasta:~] [S_base] $ ls /home/proj/stage/encrypt/
20230427_LH00188_0001_B223YYCLT3  20230508_LH00188_0003_A22522YLT3  20230824_LH00217_0004_A225CW7LT3
```


### cg backup flow-cells ✅ 
```
θ65° [henrik.stranneheim@hasta:~] [S_base] $ cg --log-level DEBUG backup flow-cells -d
Instantiating PDC api
Initialising Process with binary: /bin/dsmc
Use base call ['/bin/dsmc']
Instantiating status db
Search for flow cells ready to encrypt in /home/proj/stage/flow_cells
Found directory: /home/proj/stage/flow_cells/230912_A00621_0908_AHJWJ5DSX7
Instantiating FlowCellDirectoryData with path /home/proj/stage/flow_cells/230912_A00621_0908_AHJWJ5DSX7
Set flow cell id to AHJWJ5DSX7
Using BCL converter: dragen
Found directory: /home/proj/stage/flow_cells/230908_A01901_0231_BHJWJFDSX7
Instantiating FlowCellDirectoryData with path /home/proj/stage/flow_cells/230908_A01901_0231_BHJWJFDSX7
Set flow cell id to BHJWJFDSX7
Using BCL converter: dragen
Found directory: /home/proj/stage/flow_cells/230907_A00621_0903_AHJT27DRX3
Instantiating FlowCellDirectoryData with path /home/proj/stage/flow_cells/230907_A00621_0903_AHJT27DRX3
Set flow cell id to AHJT27DRX3
Using BCL converter: dragen
Found directory: /home/proj/stage/flow_cells/170531_D00410_0425_AHKL7LBCXY
Instantiating FlowCellDirectoryData with path /home/proj/stage/flow_cells/170531_D00410_0425_AHKL7LBCXY
Set flow cell id to AHKL7LBCXY
Using BCL converter: bcl2fastq
Found directory: /home/proj/stage/flow_cells/190927_A00689_0069_BHLYWYDSXX
Instantiating FlowCellDirectoryData with path /home/proj/stage/flow_cells/190927_A00689_0069_BHLYWYDSXX
Set flow cell id to BHLYWYDSXX
Using BCL converter: dragen
Found directory: /home/proj/stage/flow_cells/20230824_LH00217_0004_A225CW7LT3
Instantiating FlowCellDirectoryData with path /home/proj/stage/flow_cells/20230824_LH00217_0004_A225CW7LT3
Set flow cell id to A225CW7LT3
Using BCL converter: dragen
Found directory: /home/proj/stage/flow_cells/230904_A00621_0900_AHJT7VDRX3
Instantiating FlowCellDirectoryData with path /home/proj/stage/flow_cells/230904_A00621_0900_AHJT7VDRX3
Set flow cell id to AHJT7VDRX3
Using BCL converter: dragen
Found directory: /home/proj/stage/flow_cells/230824_A00187_0994_BHGW2VDRX3
Instantiating FlowCellDirectoryData with path /home/proj/stage/flow_cells/230824_A00187_0994_BHGW2VDRX3
Set flow cell id to BHGW2VDRX3
Using BCL converter: dragen
Found directory: /home/proj/stage/flow_cells/20230427_LH00188_0001_B223YYCLT3
Instantiating FlowCellDirectoryData with path /home/proj/stage/flow_cells/20230427_LH00188_0001_B223YYCLT3
Set flow cell id to B223YYCLT3
Using BCL converter: dragen
Found directory: /home/proj/stage/flow_cells/20230508_LH00188_0003_A22522YLT3
Instantiating FlowCellDirectoryData with path /home/proj/stage/flow_cells/20230508_LH00188_0003_A22522YLT3
Set flow cell id to A22522YLT3
Using BCL converter: dragen
Found directory: /home/proj/stage/flow_cells/230622_A00621_0864_BH5F5YDSX7
Instantiating FlowCellDirectoryData with path /home/proj/stage/flow_cells/230622_A00621_0864_BH5F5YDSX7
Set flow cell id to BH5F5YDSX7
Using BCL converter: dragen
Found directory: /home/proj/stage/flow_cells/230810_A00689_0860_BHGNVGDRX3
Instantiating FlowCellDirectoryData with path /home/proj/stage/flow_cells/230810_A00689_0860_BHGNVGDRX3
Set flow cell id to BHGNVGDRX3
Using BCL converter: dragen
Found directory: /home/proj/stage/flow_cells/230912_A00187_1009_AHK33MDRX3
Instantiating FlowCellDirectoryData with path /home/proj/stage/flow_cells/230912_A00187_1009_AHK33MDRX3
Set flow cell id to AHK33MDRX3
Using BCL converter: dragen
Found directory: /home/proj/stage/flow_cells/221019_A00689_0663_BH7FYHDRX2
Instantiating FlowCellDirectoryData with path /home/proj/stage/flow_cells/221019_A00689_0663_BH7FYHDRX2
Set flow cell id to BH7FYHDRX2
Using BCL converter: dragen
Found directory: /home/proj/stage/flow_cells/200207_A00689_0103_BHYG7YDSXX
Instantiating FlowCellDirectoryData with path /home/proj/stage/flow_cells/200207_A00689_0103_BHYG7YDSXX
Set flow cell id to BHYG7YDSXX
Using BCL converter: dragen
Found directory: /home/proj/stage/flow_cells/170530_ST-E00214_0154_AHJCL5ALXX
Instantiating FlowCellDirectoryData with path /home/proj/stage/flow_cells/170530_ST-E00214_0154_AHJCL5ALXX
Set flow cell id to AHJCL5ALXX
Using BCL converter: bcl2fastq
Initialising Process with binary: sbatch
Use base call ['sbatch']
Initialising Process with binary: /bin/tar
Use base call ['/bin/tar']
Initialising Process with binary: /bin/gpg
Use base call ['/bin/gpg']
SlurmAPI: Set dry run to True
Flow cell: HJWJ5DSX7 is already backed-up
Initialising Process with binary: sbatch
Use base call ['sbatch']
Initialising Process with binary: /bin/tar
Use base call ['/bin/tar']
Initialising Process with binary: /bin/gpg
Use base call ['/bin/gpg']
SlurmAPI: Set dry run to True
Flow cell: HJWJFDSX7 is already backed-up
Initialising Process with binary: sbatch
Use base call ['sbatch']
Initialising Process with binary: /bin/tar
Use base call ['/bin/tar']
Initialising Process with binary: /bin/gpg
Use base call ['/bin/gpg']
SlurmAPI: Set dry run to True
Flow cell: HJT27DRX3 is already backed-up
Initialising Process with binary: sbatch
Use base call ['sbatch']
Initialising Process with binary: /bin/tar
Use base call ['/bin/tar']
Initialising Process with binary: /bin/gpg
Use base call ['/bin/gpg']
SlurmAPI: Set dry run to True
Flow cell: HKL7LBCXY is already backed-up
Initialising Process with binary: sbatch
Use base call ['sbatch']
Initialising Process with binary: /bin/tar
Use base call ['/bin/tar']
Initialising Process with binary: /bin/gpg
Use base call ['/bin/gpg']
SlurmAPI: Set dry run to True
Flow cell: HLYWYDSXX is already backed-up
Initialising Process with binary: sbatch
Use base call ['sbatch']
Initialising Process with binary: /bin/tar
Use base call ['/bin/tar']
Initialising Process with binary: /bin/gpg
Use base call ['/bin/gpg']
SlurmAPI: Set dry run to True
Flow cell: 225CW7LT3 is already backed-up
Initialising Process with binary: sbatch
Use base call ['sbatch']
Initialising Process with binary: /bin/tar
Use base call ['/bin/tar']
Initialising Process with binary: /bin/gpg
Use base call ['/bin/gpg']
SlurmAPI: Set dry run to True
Flow cell: HJT7VDRX3 is already backed-up
Initialising Process with binary: sbatch
Use base call ['sbatch']
Initialising Process with binary: /bin/tar
Use base call ['/bin/tar']
Initialising Process with binary: /bin/gpg
Use base call ['/bin/gpg']
SlurmAPI: Set dry run to True
Flow cell: HGW2VDRX3 is already backed-up
Initialising Process with binary: sbatch
Use base call ['sbatch']
Initialising Process with binary: /bin/tar
Use base call ['/bin/tar']
Initialising Process with binary: /bin/gpg
Use base call ['/bin/gpg']
SlurmAPI: Set dry run to True
Flow cell can be backed up
Initialising Process with binary: sbatch
Use base call ['sbatch']
Initialising Process with binary: /bin/tar
Use base call ['/bin/tar']
Initialising Process with binary: /bin/gpg
Use base call ['/bin/gpg']
SlurmAPI: Set dry run to True
Flow cell can be backed up
Initialising Process with binary: sbatch
Use base call ['sbatch']
Initialising Process with binary: /bin/tar
Use base call ['/bin/tar']
Initialising Process with binary: /bin/gpg
Use base call ['/bin/gpg']
SlurmAPI: Set dry run to True
Flow cell: H5F5YDSX7 is already backed-up
Initialising Process with binary: sbatch
Use base call ['sbatch']
Initialising Process with binary: /bin/tar
Use base call ['/bin/tar']
Initialising Process with binary: /bin/gpg
Use base call ['/bin/gpg']
SlurmAPI: Set dry run to True
Flow cell: HGNVGDRX3 is already backed-up
Initialising Process with binary: sbatch
Use base call ['sbatch']
Initialising Process with binary: /bin/tar
Use base call ['/bin/tar']
Initialising Process with binary: /bin/gpg
Use base call ['/bin/gpg']
SlurmAPI: Set dry run to True
Flow cell: HK33MDRX3 is already backed-up
Initialising Process with binary: sbatch
Use base call ['sbatch']
Initialising Process with binary: /bin/tar
Use base call ['/bin/tar']
Initialising Process with binary: /bin/gpg
Use base call ['/bin/gpg']
SlurmAPI: Set dry run to True
Flow cell: H7FYHDRX2 is already backed-up
Initialising Process with binary: sbatch
Use base call ['sbatch']
Initialising Process with binary: /bin/tar
Use base call ['/bin/tar']
Initialising Process with binary: /bin/gpg
Use base call ['/bin/gpg']
SlurmAPI: Set dry run to True
Flow cell: HYG7YDSXX encryption process is not complete
Initialising Process with binary: sbatch
Use base call ['sbatch']
Initialising Process with binary: /bin/tar
Use base call ['/bin/tar']
Initialising Process with binary: /bin/gpg
Use base call ['/bin/gpg']
SlurmAPI: Set dry run to True
Flow cell: HJCL5ALXX is already backed-up
```

### cg backup fetch-flow-cells ✅ 

```
θ66° [henrik.stranneheim@hasta:~] [S_base] 21s $ cg --log-level DEBUG backup fetch-flow-cell -f 225CKHLT3 -d
Instantiating PDC api
Initialising Process with binary: /bin/dsmc
Use base call ['/bin/dsmc']
Initialising Process with binary: /bin/gpg
Use base call ['/bin/gpg']
Initialising Process with binary: /bin/tar
Use base call ['/bin/tar']
Instantiating status db
Processing flow cells: 0
Starting DSMC command:
/bin/dsmc q archive /home/proj/stage/encrypt/'*'/*225CKHLT3*.gpg
Running command /bin/dsmc q archive /home/proj/stage/encrypt/'*'/*225CKHLT3*.gpg
Dry run: process call will not be executed!!
Found archived files for PDC query: /home/proj/stage/encrypt/'*'/*225CKHLT3*.gpg
Traceback (most recent call last):
  File "/home/proj/stage/bin/miniconda3/envs/S_cg/bin/cg", line 8, in <module>
    sys.exit(base())
             ^^^^^^
  File "/home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages/click/decorators.py", line 45, in new_func
    return f(get_current_context().obj, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages/cg/cli/backup.py", line 152, in fetch_flow_cell
    retrieval_time: Optional[float] = backup_api.fetch_flow_cell(flow_cell=flow_cell)
                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages/cg/meta/backup/backup.py", line 88, in fetch_flow_cell
    archived_key: Path = self.get_archived_encryption_key_path(dsmc_output=dsmc_output)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/proj/stage/bin/miniconda3/envs/S_cg/lib/python3.11/site-packages/cg/meta/backup/backup.py", line 278, in get_archived_encryption_key_path
    encryption_key_line: str = [
                               ^
IndexError: list index out of range



Expected since we have stage in the path.

64° 15d [hiseq.clinical@hasta:~] [base] 3s 8 $ /bin/dsmc q archive /home/proj/production/encrypt/'*'/*225CKHLT3*.gpg
IBM Tivoli Storage Manager
Command Line Backup-Archive Client Interface
  Client Version 7, Release 1, Level 4.0
  Client date/time: 11/17/2023 11:10:07
(c) Copyright by IBM Corporation and other(s) 1990, 2015. All Rights Reserved.

Node Name: HASTA.SCILIFELAB.SE_CLINICAL
Session established with server BLACKHOLE: Linux/x86_64
  Server Version 8, Release 1, Level 9.300
  Server date/time: 11/17/2023 11:10:07  Last access: 11/17/2023 11:09:35

Accessing as node: SLLCLINICAL
             Size  Archive Date - Time    File - Expires on - Description
             ----  -------------------    -------------------------------
           613  B  11/16/2023 10:20:21    /home/proj/production/encrypt/20230919_LH00188_0005_A225CKHLT3/225CKHLT3.key.gpg Never Archive Date: 11/16/2023
 1,086,644,076 KB  11/16/2023 10:20:25    /home/proj/production/encrypt/20230919_LH00188_0005_A225CKHLT3/225CKHLT3.tar.gz.gpg Never Archive Date: 11/16/2023
```

**Comment by henrikstranneheim** [comment]
Deployed to production:

```
Log deploy... done.
cg, version 53.0.0
```

**Pull request merged**